### PR TITLE
Readline sync

### PR DESCRIPTION
**Pull request by jonestristand**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.

### CODE REVIEW

**Comment by jonestristand** [comment]
According to my view of the PR, it has passed the Travis CI test now.

File: readline-sync/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Type definitions for readline-sync 1.4
+// Project: https://github.com/anseki/readline-sync
+// Definitions by: Tristan Jones <https://github.com/jonestristand>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type optionType = string | number | RegExp | ((input: string) => boolean);
```
Comment (DanielRosenwasser) [inline readline-sync/index.d.ts]: Capitalize to `OptionType`

File: readline-sync/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+export function keyInPause(query?: any, options?: BasicOptions): void;
+export function keyInSelect(items: string[], query?: any, options?: BasicOptions): number;
+
+export function getRawInput(): string;
+
+// Deprecated
+export function setBufferSize(value: number): void;
+export function setEncoding(value: string): void;
+export function setMask(value: string): void;
+export function setPrint(value: ((display: string, encoding: string) => void)): void;
```
Comment (DanielRosenwasser) [inline readline-sync/index.d.ts]: Can you add a [JSDoc `@deprecated` comment](http://usejsdoc.org/tags-deprecated.html) for each of these, and also add a comment with the suggestions given in [the project's `README-Deprecated`](https://github.com/anseki/readline-sync/blob/90f173ad92d8d296a410a9b5b19df306a6ccc970/README-Deprecated.md)?

File: readline-sync/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    cd?: boolean;
+    charlist?: string;
+    min?: any;
+    max?: any;
+    confirmMessage?: any;
+    unmatchMessage?: any;
+    exists?: any;
+    isFile?: boolean;
+    isDirectory?: boolean;
+    validate?: ((path: string) => boolean | string);
```
Comment (DanielRosenwasser) [inline readline-sync/index.d.ts]: Nit: remove the outer parentheses

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Review by DanielRosenwasser — APPROVED** [review]

File: readline-sync/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    caseSensitive?: boolean;
+    keepWhitespace?: boolean;
+    encoding?: string;
+    bufferSize?: number;
+    print?: (display: string, encoding: string) => void;
+    history?: boolean;
+    cd?: boolean;
+    charlist?: string;
+    min?: any;
+    max?: any;
```
Comment (DanielRosenwasser) [inline readline-sync/index.d.ts]: `min` and `max` are both `number`s.

File: readline-sync/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    confirmMessage?: any;
+    unmatchMessage?: any;
+    exists?: any;
```
Comment (DanielRosenwasser) [inline readline-sync/index.d.ts]: Seems like `exists` really should be a `boolean`.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: readline-sync/index.d.ts (hunk 4 above)
Comment (jonestristand) [inline readline-sync/index.d.ts]: Per the readline-sync documentation:

```
min, max

Type: number or others
Default: undefined

min: A number as a minimum size of the file that is accepted.
max: A number as a maximum size of the file that is accepted.
If it is not specified or 0 is specified, the size is not checked. (A size of directory is 0.)
```

File: readline-sync/index.d.ts (hunk 5 above)
Comment (jonestristand) [inline readline-sync/index.d.ts]: Per the readline-sync package documentation:

```
exists

Type: boolean or others
Default: true

If true is specified, accept only a file or directory path that exists. If false is specified, accept only a file or directory path that does not exist.
In any other case, the existence is not checked.
```

**Comment by jonestristand** [comment]
Thanks for the review! Please see above comments. I was trying to follow the readline-sync documentation in case people are using those options for weird edge-cases that I'm not aware of. It seemed safest to follow the original design intent. If you disagree and would prefer the (more sane) types, I'll be happy to change it.

Cheers!

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
You're right about `min` and `max` being coerced, though glancing at the implementation, `exists` does seem to really expect a `boolean`. I don't think that's a blocker. If you think it'd be appropriate, you can send a PR, but thanks for this!

**Pull request merged**